src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -2200,26 +2089,116 @@ public void observe(@NonNull SCMHead head, @NonNull SCMRevision revision) throws
                 _factory.decorate(project);
                 // ok it is now up to the observer to ensure it does the actual save.
                 observer.created(project);
-                if (isAutomaticBuild(source, head, revision, null, null, listener)) {
-                    scheduleBuild(
-                            _factory,
-                            project,
-                            revision,
-                            listener,
-                            rawName,
-                            causeFactory.create(source),
-                            revisionActions
-                    );
+                doAutomaticBuilds(head, revision, rawName, project, revisionActions, null, null);
+            } finally {
+                observer.completed(encodedName);
+            }
+        }
+
+        private Action[] getRevisionActions(@NonNull SCMRevision revision, String rawName) {
+            Action[] revisionActions = new Action[0];
+            try {
+                List<Action> actions = source.fetchActions(revision, event, listener);
+                revisionActions = actions.toArray(new Action[0]);
+            } catch (IOException | InterruptedException e) {
+                printStackTrace(e, listener.error("Could not fetch metadata for revision %s of branch %s",
+                        revision, rawName));
+            }
+            return revisionActions;
+        }
+
+        private boolean isHeadActionsFetched(@NonNull SCMHead head, Branch branch, String rawName) {
```

Comment:
Not very intuitive I think for a method called _**is**Something_ to mutate the data it's given. Maybe _calculate_ or _confirm_ perhaps?